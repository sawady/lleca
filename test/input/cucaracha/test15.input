

fun f(x : Int) {
  putNum(x + 30)
  putChar(10)

  putNum(30 + x)
  putChar(10)

  z := 30
  putNum(x + z)
  putChar(10)

  putNum(z + x)
  putChar(10)
}

fun main() {
  putNum(40 + 2)
  putChar(10)

  x := 22
  y := 20
  putNum(x + y)
  putChar(10)

  x := 10
  y := 2
  f(x + y)

  x := x + 2
  f(x)
}

