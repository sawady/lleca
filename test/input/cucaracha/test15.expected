Cons(
  Function(
    f,
    Cons(
      Parameter(
        x,
        Int
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtCall(
        putNum,
        Cons(
          ExprAdd(
            ExprVar(
              x
            ),
            ExprConstNum(
              30
            )
          ),
          Nil
        )
      ),
      Cons(
        StmtCall(
          putChar,
          Cons(
            ExprConstNum(
              10
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putNum,
            Cons(
              ExprAdd(
                ExprConstNum(
                  30
                ),
                ExprVar(
                  x
                )
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  10
                ),
                Nil
              )
            ),
            Cons(
              StmtAssign(
                z,
                ExprConstNum(
                  30
                )
              ),
              Cons(
                StmtCall(
                  putNum,
                  Cons(
                    ExprAdd(
                      ExprVar(
                        x
                      ),
                      ExprVar(
                        z
                      )
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprConstNum(
                        10
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtCall(
                      putNum,
                      Cons(
                        ExprAdd(
                          ExprVar(
                            z
                          ),
                          ExprVar(
                            x
                          )
                        ),
                        Nil
                      )
                    ),
                    Cons(
                      StmtCall(
                        putChar,
                        Cons(
                          ExprConstNum(
                            10
                          ),
                          Nil
                        )
                      ),
                      Nil
                    )
                  )
                )
              )
            )
          )
        )
      )
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtCall(
          putNum,
          Cons(
            ExprAdd(
              ExprConstNum(
                40
              ),
              ExprConstNum(
                2
              )
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                10
              ),
              Nil
            )
          ),
          Cons(
            StmtAssign(
              x,
              ExprConstNum(
                22
              )
            ),
            Cons(
              StmtAssign(
                y,
                ExprConstNum(
                  20
                )
              ),
              Cons(
                StmtCall(
                  putNum,
                  Cons(
                    ExprAdd(
                      ExprVar(
                        x
                      ),
                      ExprVar(
                        y
                      )
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprConstNum(
                        10
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtAssign(
                      x,
                      ExprConstNum(
                        10
                      )
                    ),
                    Cons(
                      StmtAssign(
                        y,
                        ExprConstNum(
                          2
                        )
                      ),
                      Cons(
                        StmtCall(
                          f,
                          Cons(
                            ExprAdd(
                              ExprVar(
                                x
                              ),
                              ExprVar(
                                y
                              )
                            ),
                            Nil
                          )
                        ),
                        Cons(
                          StmtAssign(
                            x,
                            ExprAdd(
                              ExprVar(
                                x
                              ),
                              ExprConstNum(
                                2
                              )
                            )
                          ),
                          Cons(
                            StmtCall(
                              f,
                              Cons(
                                ExprVar(
                                  x
                                ),
                                Nil
                              )
                            ),
                            Nil
                          )
                        )
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    ),
    Nil
  )
)
